backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/api/impl/ESRImportBL.java

```diff
@@ -1282,8 +1316,10 @@ public void linkBankStatementLinesByPaymentIds(@NonNull final Map<PaymentId, Ban
 
 		final Set<PaymentId> paymentIds = bankStatementLineRefIdIndexByPaymentId.keySet();
 
+		final List<I_ESR_ImportLine> esrImportLines = esrImportDAO.retrieveLines(paymentIds);
+
 		final ImmutableMap<PaymentId, I_ESR_ImportLine> paySelectionLinesByPaymentId = Maps.uniqueIndex(
-				esrImportDAO.retrieveLines(paymentIds),
+				esrImportLines,
 				esrImportLine -> PaymentId.ofRepoId(esrImportLine.getC_Payment_ID()));
```

Comment:
could we run into multiple esrImportLines having the same C_Payment_ID? or are we ruling this out, e.g. with aunique contraint?
(or maybe the answer is in the code above?)

Reply:
Yes, there's a unique index: 

```sql


-- DROP INDEX public.esr_importline_payment_uq;

CREATE UNIQUE INDEX esr_importline_payment_uq
  ON public.esr_importline
  USING btree
  (c_payment_id)
  WHERE c_payment_id IS NOT NULL;

```